stackrox/stackrox
# Fix: Use Auth0 client information from an environment variable

## Description

- Changed that used client information for Auth0 development integration is provided with an environment variable
- ~Unified indenting for `deploy/common/deploy.sh` from tab/space mix to space~

## Checklist
- ~[ ] Unit test and regression tests added~ - no additional tests added
- ~[ ] Evaluated and added CHANGELOG entry if required~ - no additional tests added
- ~[ ] Determined and documented upgrade steps~ - no additional tests added
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~ - no additional tests added

If any of these don't apply, please comment below.

## Testing Performed

To speed up testing you can use existing images:
```
export MAIN_IMAGE_TAG=3.70.x-334-g964f567547
```

*Test without environment variable*
1. Teardown local deployment and install StackRox without environment variable defined:
```
teardown
./deploy/k8s/deploy-local.sh
./scripts/k8s/local-port-forward.sh
```
2. Open UI in browser https://localhost:8000/login and verify that `StackRox Dev (Auth0)` auth provider is not available.

*Test with environment variable provided*
1. Teardown local deployment and install StackRox with new environment variable defined:
```
teardown
export ROX_DEV_AUTH0_CLIENT_SECRET="<secret for Auth0 dev client>"
./deploy/k8s/deploy-local.sh
./scripts/k8s/local-port-forward.sh
```
2. Open UI in browser https://localhost:8000/login and log in using `StackRox Dev (Auth0)` auth provider.
